### ISSUE TITLE
[Bug] [Crash] IllegalStateException: ShortcutsSuggestionProvider returned duplicate suggestion IDs

### ISSUE THREAD

**Issue by csadilek** [mozilla-mobile/fenix#6855]
New crash in Sentry:
https://sentry.prod.mozaws.net/operations/fenix-nightly/issues/6753839/?environment=fenixNightly

```
java.lang.IllegalStateException: ShortcutsSuggestionProvider returned duplicate suggestion IDs
    at mozilla.components.browser.awesomebar.BrowserAwesomeBar.processProviderSuggestions$browser_awesomebar_release(BrowserAwesomeBar.kt:6)
    at mozilla.components.browser.awesomebar.BrowserAwesomeBar$queryProvidersForSuggestions$1$invokeSuspend$$inlined$forEach$lambda$1.invokeSuspend(BrowserAwesomeBar.kt:7)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:2)
    at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:15)
    at android.os.Handler.handleCallback(Handler.java:873)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loop(Looper.java:193)
    at android.app.ActivityThread.main(ActivityThread.java:6718)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Comment by csadilek** [comment]
This is a top crash now.

**Comment by csadilek** [comment]
Just found STRs:

- Switch to Locale.Germany
- Search -> Search engine settings -> Add search engine
- Select Ecosia

Ecosia is added twice causing the crash above. Since this state persists it will also crash from there one whenever the search fragment opens. Only way to get rid of the crash is by removing the search engine via the settings menu.

//cc @boek @pocmo I assume this started happening after we added Ecosia to A-C in https://github.com/mozilla-mobile/android-components/pull/5059 as it's already being added in Fenix as well. Didn't investigate further though.

**Comment by LittleAngelwings** [comment]
Can confirm @csadilek STR. Had also crashes and Ecosia twice at the search engines. After deleting Ecosia, the crashes are gone.

**Comment by mozillamarcia** [comment]
Similar signature picked up in crash stats: https://bit.ly/382CQBc

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on the latest **Nightly Build #13440604 from 12/10** using the following devices:
• Google Pixel 3a XL (Android 9)
• Huawei Mate 20 Lite (Android 8.1.0)
• Samsung Galaxy S7 (Android 7)
• OnePlus A3 (Android 6.0.1)
• LG Nexus 4 (Android 5.1.1)

►Video
![20191210_121128](https://user-images.githubusercontent.com/51314259/70521132-5f141100-1b47-11ea-907a-3fb9a6bfb641.gif)
